Doc/howto/regex.rst

```diff
@@ -1233,7 +1233,7 @@ either side.  This takes the job beyond  :meth:`!replace`'s abilities.)
 
 Another common task is deleting every occurrence of a single character from a
 string or replacing it with another single character.  You might do this with
-something like ``re.sub('\n', ' ', S)``, but :meth:`~str.translate` is capable of
+something like ``re.sub(r'\n', ' ', S)``, but :meth:`~str.translate` is capable of
```

Comment:
This change isn't strictly required, since '\n' is valid and won't raise a warning.

Reply:
This is a bit tricky.  '\n' produces a string of length 1.  r'\n' produces a string of length 2.  But re recognizes the same escapes that Python does, plus the additional ones, so the latter is also cooked down to a singles character, but later by re instead of Python itself.  I think I would be inclined to omit 'r' but mention that it can be optionally added even when the literal only has standard escapes.